### PR TITLE
Update several astronomy python libraries with new versions

### PR DESCRIPTION
**Pull request by returntoreality**
Bug: https://bugs.gentoo.org/701454

### CODE REVIEW

**Comment by stkw0** [comment]
I think we should use EAPI=7 instead of 6.

**Comment by stkw0** [comment]
Did the tests ran successfully in astropy?

**Comment by returntoreality** [comment]
@stkw0 So I got the tests to pass. I had to add several packages, as the astropy tests depend on them to run and update the dependencies. I don't really know if I did everything correct with the metadata.xml files, as this is the first time I touched them. If the maintainer needs to be different (or if I should place me as maintainer) I will change them as needed.
It seems the packages were just deleted which is unfortunate. Would this have any impact on the acceptance of this pull request? I would rebase so these packages would be readded.

**Comment by heroxbd** [comment]
Thank you very much @returntoreality.

**Comment by returntoreality** [comment]
@stkw0 I've rebased my changes on the current tree, everything is now EAPI=7, astropy tests work (although I had to disable a test, that fails when the timezone-data is too old. It was discussed upstream to maybe remove this test as it has other issues https://github.com/astropy/astropy/pull/9720#issuecomment-561437703 ).
I'm not sure if I did everything correctly with the metadata.xml files. For the new packages I put the sci-astronomy or the python team, where I thought it made sense. Should I add myself as well? Is there some process that has to be followed before packages can be added to these teams?

**Comment by stkw0** [comment]
I'm not a developer, so I'm not sure if it's okay to add them directly, although with flask packages I was told to add the python team and I suppose that the same applies here. In any case, if you are interested in maintaining those packages take a look at https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers and ask on the IRC channel for those questions.

**Comment by returntoreality** [comment]
I see, thanks for the info! I will update the metadata.xml files then and add myself and the proxy maintainer team as maintainers as well.

**Comment by stkw0** [comment]
`repoman -dx full` is reporting warnings for a couple of packages. You should take a look. Also, before adding yourself to the maintainers list a dev must open an issue in bugzilla (if I remember correctly). Ask on the proxy-maint IRC channel about it. 

I'm also interested in resurrecting those packages, maybe we can work together.

**Comment by returntoreality** [comment]
> `repoman -dx full` is reporting warnings for a couple of packages. You should take a look. Also, before adding yourself to the maintainers list a dev must open an issue in bugzilla (if I remember correctly). Ask on the proxy-maint IRC channel about it.
So I removed all ~amd64-linux keywords where it caused repoman to complain (mainly because no dev-lang/python package has the ~amd64-linux keyword. According to GLEP 53 this should be the same as ~amd64, but apparently this is not the case). But this also causes KEYWORDS.dropped messages from repoman, for the version bumps (e.g. dev-python/pytest-arraydiff). 
I tried to find some documentation that states whether the ~amd64-linux keyword should or should not be used. The ~amd64-linux keyword should be equivalent to ~amd64 (according to GLEP 53) and no issues are raised for ~amd64 (maybe this is a repoman bug?).
Aside from the KEYWORDS.dropped complaints, the only one left is sci-astronomy/wcslib, because of a `emake -j1` (the comment in the original ebuild was "nasty makefile, debugging means probably full rewrite"). 
Regarding the maintainer bug: I read in the Proxy Maint user guide: "Once your first contribution is merged, we will create the maintainer bug for you". I will ask in the irc to be sure.
> 
> I'm also interested in resurrecting those packages, maybe we can work together.
Sure, that would be great. I could add you to the metadata.xml as well, but I don't know if this is the correct procedure. Maybe it would be best if you create a pull request adding yourself to these packages after this pull request is merged? Maybe a developer could answer this.

**Comment by heroxbd** [comment]
> > `repoman -dx full` is reporting warnings for a couple of packages. You should take a look. Also, before adding yourself to the maintainers list a dev must open an issue in bugzilla (if I remember correctly). Ask on the proxy-maint IRC channel about it.
> 
> So I removed all ~amd64-linux keywords where it caused repoman to complain (mainly because no dev-lang/python package has the ~amd64-linux keyword. According to GLEP 53 this should be the same as ~amd64, but apparently this is not the case). But this also causes KEYWORDS.dropped messages from repoman, for the version bumps (e.g. dev-python/pytest-arraydiff).
> I tried to find some documentation that states whether the ~amd64-linux keyword should or should not be used. The ~amd64-linux keyword should be equivalent to ~amd64 (according to GLEP 53) and no issues are raised for ~amd64 (maybe this is a repoman bug?).

The `amd-linux` is for Gentoo Prefix rpath.

File: dev-python/asdf/asdf-2.5.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		python_setup
+		PYTHONPATH="${BUILD_DIR}"/lib \
+				  esetup.py build_sphinx --no-intersphinx
+	fi
+}
+
+python_test() {
+	pytest -v
+}
+
```
Comment (heroxbd) [inline dev-python/asdf/asdf-2.5.1.ebuild]: Can this be replaced with distutils_enable_tests ? (https://devmanual.gentoo.org/eclass-reference/distutils-r1.eclass)

File: dev-python/astropy/astropy-4.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	if use doc; then
+		VARTEXFONTS="${T}"/fonts MPLCONFIGDIR="${T}" PYTHONPATH="${BUILD_DIR}"/lib \
+				   esetup.py build_docs --no-intersphinx
+		HTML_DOCS=( docs/_build/html/. )
+	fi
+}
+
+python_test() {
+	esetup.py test
+}
```
Comment (heroxbd) [inline dev-python/astropy/astropy-4.0.ebuild]: The same, can it be simplified by `distutils_enable_tests`?

File: dev-python/gwcs/gwcs-0.12.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		python_setup
+		PYTHONPATH="${BUILD_DIR}"/lib \
+				  esetup.py build_sphinx --no-intersphinx
+	fi
+}
+
+python_test() {
+	esetup.py test
+}
+
```
Comment (heroxbd) [inline dev-python/gwcs/gwcs-0.12.0.ebuild]: `distutils_enable_tests()`?

**Comment by heroxbd** [comment]
@returntoreality, your work is very much appreciated.  But unfortunately, no Gentoo developer showed interest in maintaining them.  On 2019-12-28, many of the packages were removed from Gentoo repository completely.  Before adding them back to the gentoo repository, we should seriously ask ourselves whether we are committed to maintain astronomy python.

If you think you are going to take this glorious responsibility, I suggest that we move to the Science overlay.  You've shown your ability for ebuild development in this PR.  I don't think it would take long for you to be granted write access to the science overlay.  If you want to contribute more, you can seek to become a developer after gaining experience in the science overlay to reintroduce the packages back into Gentoo main repository.

**Comment by stkw0** [comment]
@heroxbd thanks. I'm moving some packages to gentoo/sci.

**Comment by ajakk** [comment]
Seems like there's some consensus that ::gentoo is not the best place for these packages. Thank you for your work, nonetheless!

**Pull request closed without merging**